File: add-sections.py
import json
import random
import string

# Read current db.json
with open('db.json', 'r', encoding='utf-8') as f:
    db = json.load(f)

# Helper function to create a hero section
def create_hero_section(page_title, page_id):
    random_id = ''.join(random.choices(string.ascii_lowercase + string.digits, k=12))
    
    return {
        "id": f"hero-{random_id}",
        "type": "hero",
        "content": {
            "en": {
                "title": f"Welcome to {page_title['en']}",
                "subtitle": f"Discover our {page_title['en'].lower()} programs and initiatives",
                "buttonText": "Learn More",
                "buttonLink": "#"
            },
            "ar": {
                "title": f"مرحباً بك في {page_title['ar']}",
                "subtitle": f"اكتشف برامجنا ومبادراتنا",
                "buttonText": "اعرف المزيد",
                "buttonLink": "#"
            },
            "ku": {
                "title": f"بەخێربێیت بۆ {page_title['ku']}",
                "subtitle": "بەرنامە و دەستپێشخەریەکانمان بدۆزەرەوە",
                "buttonText": "زیاتر بزانە",
                "buttonLink": "#"
            }
        },
        "settings": {
            "backgroundType": "color",
            "backgroundColor": "#1e40af",
            "textColor": "#ffffff",
            "backgroundImage": "",
            "paddingTop": "100px",
            "paddingBottom": "100px"
        },
        "order": 1
    }

# Add hero section to pages without sections
count = 0
for page in db['pages']:
    if not page.get('sections') or len(page['sections']) == 0:
        page['sections'] = [create_hero_section(page['title'], page['id'])]
        count += 1

# Write updated db.json
with open('db.json', 'w', encoding='utf-8') as f:
    json.dump(db, f, ensure_ascii=False, indent=2)

print(f"Done! Added sections to {count} pages!")

File: add-sections.js
const fs = require('fs');

// Read current db.json
const db = JSON.parse(fs.readFileSync('db.json', 'utf8'));

// Helper function to create a hero section
function createHeroSection(pageTitle) {
  return {
    id: `hero-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`,
    type: 'hero',
    content: {
      en: {
        title: `Welcome to ${pageTitle.en}`,
        subtitle: `Discover our ${pageTitle.en.toLowerCase()} programs and initiatives`,
        buttonText: 'Learn More',
        buttonLink: '#'
      },
      ar: {
        title: `مرحباً بك في ${pageTitle.ar}`,
        subtitle: `اكتشف برامجنا ومبادراتنا في ${pageTitle.ar}`,
        buttonText: 'اعرف المزيد',
        buttonLink: '#'
      },
      ku: {
        title: `بەخێربێیت بۆ ${pageTitle.ku}`,
        subtitle: `بەرنامە و دەستپێشخەریەکانمان بدۆزەرەوە`,
        buttonText: 'زیاتر بزانە',
        buttonLink: '#'
      }
    },
    settings: {
      backgroundType: 'color',
      backgroundColor: '#1e40af',
      textColor: '#ffffff',
      backgroundImage: '',
      paddingTop: '100px',
      paddingBottom: '100px'
    },
    order: 1
  };
}

// Add hero section to pages without sections
db.pages.forEach(page => {
  if (!page.sections || page.sections.length === 0) {
    page.sections = [createHeroSection(page.title)];
  }
});

// Write updated db.json
fs.writeFileSync('db.json', JSON.stringify(db, null, 2), 'utf8');
console.log('✅ Added sections to all pages!');
